pygfx/utils/text/_fontfinder.py

```diff
@@ -0,0 +1,260 @@
+        # that's that - whatever is reading is can keep doing so,
+        # because this is simply a new inode with the same name. On
+        # Windows, however, the replace will fail if something else has
+        # the file open. So we wait and try again, until we time out
+        # and give up.
+        etime = time.time() + 10
+        while time.time() < etime:
+            try:
+                os.replace(filename2, filename)
+            except OSError:
+                time.sleep(0.1)
+            else:
+                break
+        else:
+            logger.error("Failed to write font cache")
+            try:
+                os.remove(filename2)
+            except Exception:
+                pass
+
+    return list(cache["files"].keys())
+
+
+def find_system_fonts():
+    """Find fonts on the file system. Returns (dir_paths, file_paths).
+    This function might be relatively slow.
+    """
+    dirs = get_system_font_directories()
+    dir_paths = set()
+    file_paths = set()
+    for d in dirs:
+        new_dirs, new_files = find_fonts_from_dir(d, True)
+        dir_paths.update(new_dirs)
+        file_paths.update(new_files)
+
+    return dir_paths, file_paths
+
+
+# %% OS-specific logic
+
+
+def get_system_font_directories():
+    # Simple triage, easy to replace in tests.
+    if sys.platform.startswith("win"):
+        return get_windows_font_directories()
+    elif sys.platform.startswith("darwin"):
+        return get_osx_font_directories()
+    else:
+        return get_unix_font_directories()
+
+
+def get_windows_font_directories():
+    import winreg
+
+    dirs = []
+    dirs.extend(WinFontDirs)
+
+    # Get win32 font directory from the registry
+    ms_folders = r"Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders"
+    try:
+        with winreg.OpenKey(winreg.HKEY_CURRENT_USER, ms_folders) as user:
+            dirs.append(winreg.QueryValueEx(user, "Fonts")[0])
+    except OSError:
+        dirs.append(os.path.join(os.environ["WINDIR"], "Fonts"))
+
+    return [os.path.abspath(d) for d in dirs if os.path.isdir(d)]
+
+
+def get_osx_font_directories():
+    dirs = []
+    dirs.extend(X11FontDirectories)
+    dirs.extend(OSXFontDirectories)
+    return [os.path.abspath(d) for d in dirs if os.path.isdir(d)]
+
+
+def get_unix_font_directories():
+    # MPL also calls out to fontconfig (fc-list) to get a list of fonts.
+    # But if we assume that our list of possible directories is
+    # complete, this should not be necessary. Let's see how it goes.
+    dirs = []
+    dirs.extend(X11FontDirectories)
+    return [os.path.abspath(d) for d in dirs if os.path.isdir(d)]
+
+
+try:
+    HOME = os.path.expanduser("~")
+except Exception:  # Exceptions thrown by home() are not specified...
+    HOME = "/home"  # Just an arbitrary path
+
+WinFontDirs = [
+    os.path.join(HOME, "AppData/Local/Microsoft/Windows/Fonts"),
+    os.path.join(HOME, "AppData/Roaming/Microsoft/Windows/Fonts"),
+]
```

Comment:
I would recommend extending this with the common user folder, and not relying on HOME expansion:

```suggestion
WinFontDirs = [
    os.path.join(os.environ["LOCALAPPDATA"], "Microsoft/Windows/Fonts"),
    os.path.join(os.environ["APPDATA"], "Microsoft/Windows/Fonts"),
    os.path.join(os.environ["ALLUSERSPROFILE"], "AppData/Local/Microsoft/Windows/Fonts"),
    os.path.join(os.environ["ALLUSERSPROFILE"], "AppData/Roaming/Microsoft/Windows/Fonts"),
]
```